core/src/main/java/software/amazon/awssdk/core/runtime/transform/PathMarshaller.java

```diff
@@ -24,7 +24,7 @@
 import software.amazon.awssdk.utils.http.SdkHttpUtils;
 
 @SdkProtectedApi
-public final class PathMarshallers {
+public abstract class PathMarshaller {
```

Comment:
Can this still be an interface? Or does it have something to do with the private static classes?

Reply:
Nothing can be private in an interface (before Java 9).